File: 250901/금 채굴하기/gold-mining.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] grid = new int[n][n];
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                grid[i][j] = sc.nextInt();
        // Please write your code here.

        int max = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k <= 2 * (n - 1); k++) {
                    int sum = 0;

                    for (int r = 0; r < n; r++) {
                        for (int c = 0; c < n; c++) {
                            if (Math.abs(i - r) + Math.abs(j - c) <= k) {
                                sum += grid[r][c];
                            }
                        }
                    }
                    if (sum * m >= Math.pow(k,2) + Math.pow((k+1), 2)) {
                        max = Math.max(max, sum);
                    }
                }                
            }
        }

        System.out.println(max);



    }
}

File: 260317/금 채굴하기/gold-mining.py
n, m = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]

# Please write your code here.


max_count = 0
for r in range(n):
    for c in range(n):
        for k in range(n + 1):
            count = 0
            for rr in range(n):
                for cc in range(n):
                    dis = abs(r - rr) + abs(c - cc)
                    if grid[rr][cc] == 1 and dis <= k :
                        count += 1
            
            if m * count >= pow(k, 2) + pow(k + 1, 2):
                max_count = max(max_count, count)
            
            
print(max_count)


